build/commands/lib/config.js

```diff
@@ -92,8 +92,18 @@ const parseExtraInputs = (inputs, accumulator, callback) => {
 }
 
 const Config = function () {
-  this.sardineClientId = getNPMConfig(['sardine_client_id']) || ''
-  this.sardineClientSecret = getNPMConfig(['sardine_client_secret']) || ''
+  // Populate `.env` file to custom object to avoid leakage
+  const dotenvConfig = {}
+  require('dotenv').config({ processEnv: dotenvConfig });
+  const getDotenvConfig = (k) => {
+    if (k.length == 1) {
+        return dotenvConfig[k[0]];
```

Comment:
tried this on Windows, it almost works, but `gn` booleans are failing.

I think we should simplify this PoC to this:
```
--- a/build/commands/lib/config.js
+++ b/build/commands/lib/config.js
@@ -57,6 +57,12 @@ const getNPMConfig = (key, default_value = undefined) => {
   if (!NpmConfig) {
     const list = run(npmCommand, ['config', 'list', '--json', '--userconfig=' + path.join(rootDir, '.npmrc')])
     NpmConfig = JSON.parse(list.stdout.toString())
+    require('dotenv').config({ processEnv: NpmConfig, override: true })
+    for (let [key, value] of Object.entries(NpmConfig)) {
+      if (value === 'true' || value === 'false') {
+        NpmConfig[key] = value === 'true'
+      }
+    }
   }
```

1. please note added `override: true` so `.env` values are preferred over `.npmrc`.
2. the `NpmConfig[key] = value === 'true'` is to fix `gn` boolean values.

when we migrate CI, we can cleanup this and remove all `NPMConfig` mentions.

Reply:
ooh, much better approach, thanks! Updated.